### PR TITLE
[dynamic] Support delete on sdkv1 based providers

### PR DESCRIPTION
**Pull request by iwahbe**
Explicitly set `PlannedState: null` instead of leaving it empty.

As described by https://github.com/hashicorp/terraform-plugin-framework/blob/ce2519cf40d45d28eebd81776019e68d1bddca6f/internal/fwserver/server_applyresourcechange.go#L63, PF allows Null (not just nil) values to be used to send a delete. SDKv1 providers *always* attempt to deserialize planned state. This is fine if it's null, but hits a nil pointer panic when it is nil.

Fixes https://github.com/pulumi/pulumi-terraform-provider/issues/36

### CODE REVIEW

File: pkg/pf/tfbridge/provider_delete.go [hunk 1]
```diff
@@ -52,13 +53,17 @@ func (p *provider) DeleteWithContext(
 		return resource.StatusOK, err
 	}
 
+	plannedState, err := tfprotov6.NewDynamicValue(tfType, tftypes.NewValue(tfType, nil))
+	contract.AssertNoErrorf(err, "nil is always a valid value to marshal to a dynamic state")
```
Comment (VenelinMartinov) [inline pkg/pf/tfbridge/provider_delete.go:57]: Did you mean dynamic value?

File: pkg/pf/tfbridge/provider_delete.go [hunk 2]
```diff
@@ -52,13 +53,17 @@ func (p *provider) DeleteWithContext(
...
 	// terraform-plugin-framework recognizes PlannedState=nil ApplyResourceChangeRequest request as DELETE.
 	//
 	//nolint:lll // See
 	// https://github.com/hashicorp/terraform-plugin-framework/blob/ce2519cf40d45d28eebd81776019e68d1bddca6f/internal/fwserver/server_applyresourcechange.go#L63
 	req := tfprotov6.ApplyResourceChangeRequest{
-		TypeName:   rh.terraformResourceName,
-		PriorState: priorState,
+		TypeName:     rh.terraformResourceName,
+		PriorState:   priorState,
+		PlannedState: &plannedState,
```
Comment (VenelinMartinov) [inline pkg/pf/tfbridge/provider_delete.go:66]: How does this affect PF? I'd love to see some tests there to prove it doesn't.

**Review by VenelinMartinov — APPROVED** [review]

**Review by VenelinMartinov — CHANGES_REQUESTED** [review]
Approved by accident - is PF already tested here?

File: pkg/pf/tfbridge/provider_delete.go (hunk 2 above)
Comment (t0yv0) [reply pkg/pf/tfbridge/provider_delete.go:66]: I'm pretty sure deletion lifecycle is covered for PF, but let me poke in the cov reports.

**Comment by t0yv0** [comment]
It's covered per codecov. I think this can go.

**Review by t0yv0 — APPROVED** [review]
Thank you!

**Comment by VenelinMartinov** [comment]
Is codecov a sufficient pointer for coverage? I thought we agreed that is not the case.

**Comment by t0yv0** [comment]
https://github.com/pulumi/pulumi-terraform-bridge/blob/master/pkg/pf/tests/integration/program_test.go#L48 at the very least covers the Delete lifecycle method

**Review by VenelinMartinov — APPROVED** [review]
seems to be covered by tests

File: pkg/pf/tfbridge/provider_delete.go (hunk 1 above)
Comment (iwahbe) [reply pkg/pf/tfbridge/provider_delete.go:57]: No, I meant the `nil` in `tftypes.NewValue(tfType, nil)`.

Comment (VenelinMartinov) [reply pkg/pf/tfbridge/provider_delete.go:57]: The error says "dynamic state" not "dynamic value" - what is dynamic state? @iwahbe

Comment (iwahbe) [reply pkg/pf/tfbridge/provider_delete.go:57]: Implicitly, a dynamic value that is treated as state, in this case planned state.

**Pull request merged**